design/0001-frr.md

```diff
@@ -310,3 +310,58 @@ releases is proposed:
    MetalLB is deprecated from this point onwards. FRR is no longer
    limitied by feature parity with the current native implementation of BGP and
    additional functionality may be developed.
+
+## Test Coverage
+
+The current e2e tests setup a bgp peer to all the nodes inside an FRR container, and try to hit the service.
+
+In order to guarantee parity between the two implementations, the coverage of the e2e tests must be
+extended.
+
+### Test strategy
+
+Every time we test an exposed service, we must verify that:
+
+* the service is reacheable from outside
+* the nodes advertising the service ip are the expected ones
+* all the nodes are paired with the expected bgp peers
+
+#### Changing configuration
+
+Every time the configuration is changed before testing the new scenario, we
+may need to wait an arbitrary time or to run the checks in an eventually loop,
+since there is no feedback of what configuration is being consumed.
+
+Another possible option is to remove the configuration completely and
+verify that hitting the service does not work anymore. This should ensure that
+it's possible to apply a new configuration.
+
+#### Coverage
+
+We must cover the following scenarios:
+
+* Cluster Traffic Policy
+* Local Traffic Policy
+* One Peer
+* Multiple Peers
+* Configuration Change (adding / removing new pools, adding / removing bgp peers)
+* Failover: killing one node and see that we are still able to reach the service
```

Comment:
We should probably check that keepalives are received?

Reply:
Isn't checking the state of the peers / routes enough to verify that the new state is adjusted?

Reply:
Yes, you are probably right.